File: CSharp/SubscriptionApi/SubscriptionApi/Authorization/AssetQueryResolution.cs
using System.Collections.Generic;

namespace SubscriptionApi.Authorization
{
  public interface AssetQueryResolution
  {
    List<string> RetrieveAssetsByHardwareName(string name);
    List<string> RetrieveAssetsByUserName(string name);
    List<string> RetrieveAssetsByOrganizationalUnitName(string name);
  }
}

File: Java/factories/_01_no_factories/src/main/java/Authorization/AssetQueryResolution.java
package authorization;

import java.util.List;

public interface AssetQueryResolution {
    List<String> retrieveAssetsByHardwareName(String name);

    List<String> retrieveAssetsByUserName(String name);

    List<String> retrieveAssetsByOrganizationalUnitName(String name);
}
